### PR TITLE
Add StdScaler for time series Transformer model

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

- [x] Add `loc` and `scale` outputs from current scalers and use both as static real-valued covariates
- [ ] double check training/inference works as usual
- [ ] add the StdScaler

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/time_series_transformer/modeling_time_series_transformer.py [hunk 1]
```diff
@@ -443,9 +446,12 @@ class Seq2SeqTimeSeriesModelOutput(ModelOutput):
...
             Attentions weights of the encoder, after the attention softmax, used to compute the weighted average in the
             self-attention heads.
+        loc: (`torch.FloatTensor` of shape `(batch_size,)`, *optional*):
+            Shift values of each time series' context window which is used to give the model inputs of the same
+            magnitude and then used to shift back to the original magnitude.
         scale: (`torch.FloatTensor` of shape `(batch_size,)`, *optional*):
             Scaling values of each time series' context window which is used to give the model inputs of the same
-            magnitude and then used to rescale to the original scale.
+            magnitude and then used to rescale back to the original magnitude.
         static_features: (`torch.FloatTensor` of shape `(batch_size, feature size)`, *optional*):
```
Comment (sgugger) [inline src/transformers/models/time_series_transformer/modeling_time_series_transformer.py:455]: There should be no colon after the name of the arg, only at the end of the line.

File: src/transformers/models/time_series_transformer/modeling_time_series_transformer.py [hunk 2]
```diff
@@ -510,9 +517,12 @@ class Seq2SeqTimeSeriesPredictionOutput(ModelOutput):
...
```
Comment (sgugger) [inline src/transformers/models/time_series_transformer/modeling_time_series_transformer.py:526]: Same here.

File: src/transformers/models/time_series_transformer/modeling_time_series_transformer.py [hunk 3]
```diff
@@ -1834,7 +1849,9 @@ def forward(
         params = None
         if future_values is not None:
             params = self.output_params(outputs[0])  # outputs.last_hidden_state
-            distribution = self.output_distribution(params, outputs[-2])  # outputs.scale
+            distribution = self.output_distribution(
+                params, loc=outputs[-3], scale=outputs[-2]
+            )  # outputs.loc, outputs.scale
```
Comment (sgugger) [inline src/transformers/models/time_series_transformer/modeling_time_series_transformer.py:1854]: Put the comment above so the formatting tools don't break the line.

**Review by sgugger — COMMENTED** [review]
Thanks for adding this! I have a few comments.

**Comment by kashif** [comment]
thanks! will add... moving this to draft for now as i have to check this before adding

**Pull request closed without merging**